From: uazo <uazo@users.noreply.github.com>
Date: Thu, 9 Jun 2022 19:45:03 +0000
Subject: Remove segmentation platform

Also fixes a crash on startup.

Original License: GPL-2.0-or-later - https://spdx.org/licenses/GPL-2.0-or-later.html
License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html

Change-Id: Ief303535f95c44b1bfaa56973fe37e0f01dbc61f
---
 .../chrome_browser_main_extra_parts_segmentation_platform.cc  | 4 ++++
 components/segmentation_platform/public/features.cc           | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/chrome/browser/segmentation_platform/chrome_browser_main_extra_parts_segmentation_platform.cc b/chrome/browser/segmentation_platform/chrome_browser_main_extra_parts_segmentation_platform.cc
--- a/chrome/browser/segmentation_platform/chrome_browser_main_extra_parts_segmentation_platform.cc
+++ b/chrome/browser/segmentation_platform/chrome_browser_main_extra_parts_segmentation_platform.cc
@@ -12,11 +12,13 @@
 #include "components/segmentation_platform/public/segmentation_platform_service.h"
 
 void ChromeBrowserMainExtraPartsSegmentationPlatform::PreCreateThreads() {
+  if ((true)) return;
   segmentation_platform::LocalStateHelper::GetInstance().Initialize(
       g_browser_process->local_state());
 }
 
 void ChromeBrowserMainExtraPartsSegmentationPlatform::PreProfileInit() {
+  if ((true)) return;
   segmentation_platform::UkmDatabaseClientHolder::GetClientInstance(nullptr)
       .PreProfileInit(
           /*in_memory_database=*/false);
@@ -25,6 +27,7 @@ void ChromeBrowserMainExtraPartsSegmentationPlatform::PreProfileInit() {
 void ChromeBrowserMainExtraPartsSegmentationPlatform::PostProfileInit(
     Profile* profile,
     bool is_initial_profile) {
+  if ((true)) return;
   if (!profile || profile->IsOffTheRecord())
     return;
 
@@ -36,6 +39,7 @@ void ChromeBrowserMainExtraPartsSegmentationPlatform::PostProfileInit(
 }
 
 void ChromeBrowserMainExtraPartsSegmentationPlatform::PostMainMessageLoopRun() {
+  if ((true)) return;
   segmentation_platform::UkmDatabaseClientHolder::GetClientInstance(nullptr)
       .PostMessageLoopRun();
 }
diff --git a/components/segmentation_platform/public/features.cc b/components/segmentation_platform/public/features.cc
--- a/components/segmentation_platform/public/features.cc
+++ b/components/segmentation_platform/public/features.cc
@@ -267,4 +267,7 @@ BASE_FEATURE(kSegmentationPlatformFedCmUser,
              "SegmentationPlatformFedCmUser",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformFeature);
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformFeedSegmentFeature);
+SET_CROMITE_FEATURE_DISABLED(kSegmentationPlatformAdaptiveToolbarV2Feature);
 }  // namespace segmentation_platform::features
-- 

